apps/toys/DragDrop/src/dragdrop/DragDropDuke.java

```diff
@@ -0,0 +1,158 @@
+import java.net.URL;
+
+public class DragDropDuke extends Application {
+    private Image image;
+    private ProgressBar progressBar;
+    private ImageView imageView;
+    private Label posLabel;
+
+    @Override
+    public void start(Stage stage) {
+        progressBar = new ProgressBar();
+        progressBar.prefWidthProperty().bind(stage.widthProperty().multiply(0.80d));
+
+        new Thread(this::download).start();
+
+        Text text = new Text("Instructions:\n" +
+                "1. Wait for the image download.\n" +
+                "2. Drag image and notice it's a transparent png.\n" +
+                "3. Notice Position changes from \"Mouse Move\" to \"Mouse Drag\".\n" +
+                "4. Drag over and drop on the tomato square (should change colors).\n" +
+                "5. Open another instance of this Window and drag from one\ninstance to another.\n" +
+                "6. While dragging image, press ALT+F4 (should stop Drag).");
+
+        imageView = new ImageView(image);
+        imageView.setFitHeight(200D);
+        imageView.setPreserveRatio(true);
+
+        Pane pane = new Pane();
+        pane.setPrefSize(200, 200);
+        pane.setBackground(Background.fill(Color.TOMATO));
+        HBox hBox = new HBox(imageView, pane);
+        hBox.setAlignment(Pos.CENTER);
+
+        posLabel = new Label();
+        VBox vBox = new VBox(text, posLabel, progressBar, hBox);
+        vBox.setSpacing(5.0);
+        vBox.setAlignment(Pos.CENTER);
+        stage.setScene(new Scene(vBox, 480, 480));
+        stage.setTitle("Drag & Drop Duke");
+
+        imageView.setOnDragDetected(event -> {
+            ClipboardContent content = new ClipboardContent();
+            content.putImage(image);
+
+            Dragboard dragboard = imageView.startDragAndDrop(TransferMode.ANY);
+            dragboard.setContent(content);
+            dragboard.setDragViewOffsetX(100);
+            dragboard.setDragViewOffsetY(100);
+            dragboard.setDragView(image);
+        });
+
+        stage.getScene().setOnMouseMoved(e -> showPos("Mouse Move", e.getSceneX(), e.getSceneY()));
+        stage.getScene().setOnDragOver(e -> showPos("Mouse Drag", e.getSceneX(), e.getScreenY()));
+
+        pane.setOnDragOver(e -> {
+            e.acceptTransferModes(TransferMode.ANY);
+            pane.setBackground(Background.fill(Color.CADETBLUE));
+        });
+        pane.setOnDragDropped(e -> pane.setBackground(Background.fill(Color.LAWNGREEN)));
+        stage.show();
+    }
+
+    private void showPos(String prefix, double x, double y) {
+        posLabel.setText(String.format("%s -> x: %f, y: %f", prefix, x, y));
+    }
+
+    private void download() {
+        try {
+            subDownload();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void subDownload() throws IOException {
+        URL url = new URL("http://cr.openjdk.java.net/~jeff/Duke/png/Hips.png");
```

Comment:
I think, if possible, tests should be as independent from online-hosted content as possible (and in this case it seems like we only need an Image). I'd consider either adding Duke image to the repo, or (if adding a .png is not a good solution either - @kevinrushforth would have to confirm) generating a Test image.